opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -125,12 +145,9 @@ impl<T: Number<T>> Sum<T> {
         s_data.is_monotonic = self.monotonic;
         s_data.data_points.clear();
 
-        let mut values = match self.value_map.values.write() {
-            Ok(v) => v,
-            Err(_) => return (0, None),
-        };
-
-        let n = values.len() + 1;
+        // Max number of data points need to account for the special casing
+        // of the no attribute value + overflow attribute.
+        let n = self.value_map.count.load(Ordering::SeqCst) + 2;
```

Comment:
I'm not sure if this calculation is important for correctness of the export. If it is, then we should do this under the `self.value_map.values.write()` lock. Otherwise, we read a certain value here and before we reserve capacity for data points, some update thread(s) could enter the write lock and increment the count further.

Reply:
It does not look important for correctness. Even if it is off by few, it just results in vec resizing in the middle of collect.
We can revisit this.